File: java/src/main/java/vsdk/toolkit/raycasting/stochasticRaytracing/LightSourceTable.java
package vsdk.toolkit.raycasting.stochasticRaytracing;

import vsdk.toolkit.environment.geometry.elements.Patch;

public class LightSourceTable {
    public Patch patch;
    public double flux;

    public LightSourceTable() {
        patch = null;
        flux = 0.0;
    }

    public LightSourceTable(Patch patch, double flux) {
        this.patch = patch;
        this.flux = flux;
    }
}

File: typescript/src/main/vsdk/toolkit/raycasting/stochasticRaytracing/LightSourceTable.ts
import { Patch } from "../../environment/geometry/elements/Patch";

export class LightSourceTable {
  public patch: Patch | null;
  public flux: number;

  public constructor(patch: Patch | null = null, flux = 0.0) {
    this.patch = patch;
    this.flux = flux;
  }
}
